### DISCUSSION TITLE
Entity with Entity collide event

### DISCUSSION THREAD

**Discussion by arnhav** [PaperMC/Paper#10557] [Ideas]
### Is your feature request related to a problem?

I am currently writing a plugin that simulates a cavalry "charge" of sorts. Basically, while riding a horse, if you run into an entity, that entity gets damaged and knocked back a bit.

I found the VehicleEntityCollisionEvent but apparently that only fires for boats and minecarts.

### Describe the solution you'd like.

Either:
- Expanding the VehicleEntityCollisionEvent to fire for horses
- A new EntityCollideWithEntityEvent (not too sure on the naming here lol)

### Describe alternatives you've considered.

Currently, for this I have to use the EntityMoveEvent and check the bounding boxes there.

**Comment by JMBoy01** [comment]
Hi, I might be able to help with expanding the event. I would also suggest to do this for all rideable animals such as lamma, pig, strider, donkey, skeleton and zombie horse.

**Comment by codebycam** [comment]
Relevant PR: https://github.com/PaperMC/Paper/pull/9998

**Comment by notTamion** [comment]
new pr #11906
